packages/console/src/monaco/MonacoProviders.tsx

```diff
@@ -93,30 +130,34 @@ class MonacoProviders extends PureComponent<
   }
 
   componentDidMount(): void {
-    const { language } = this.props;
+    const { language, session } = this.props;
 
     this.registeredCompletionProvider = monaco.languages.registerCompletionItemProvider(
-      language ?? '',
+      language,
       {
         provideCompletionItems: this.handleCompletionRequest,
-        triggerCharacters: ['.'],
+        triggerCharacters: ['.', '"', "'"],
       }
     );
 
-    this.registeredSignatureProvider = monaco.languages.registerSignatureHelpProvider(
-      language ?? '',
-      {
-        provideSignatureHelp: this.handleSignatureRequest,
-        signatureHelpTriggerCharacters: ['(', ','],
-      }
-    );
+    if (session.getSignatureHelp) {
```

Comment:
The default rules allow it because function does not have a falsy context. The rule defaults are to avoid `if (str)` if str is nullable since that condition covers both null and empty string

If you want to open a ticket to change it, the scenarios that are allowed by default are `allowString`, `allowNumber`, and `allowNullableObject`

https://typescript-eslint.io/rules/strict-boolean-expressions/#options